tools/docker-go

```diff
@@ -56,14 +56,15 @@ parse_args "${@}"
 
 # Go accepts both lower and uppercase proxy variables, pass both through.
 proxy_env=( )
-for i in http_proxy https_proxy no_proxy HTTP_PROXY HTTPS_PROXY NO_PROXY; do
-   if [ -n "${!i}" ]; then
-      proxy_env[${#proxy_env[@]}]="--env=$i=${!i}"
-   fi
+for i in \
+    http_proxy https_proxy no_proxy HTTP_PROXY HTTPS_PROXY NO_PROXY \
+    goproxy gonoproxy goprivate GOPROXY GONOPROXY GOPRIVATE ; do
```

Comment:
If a `go` binary is present on the host where this is running, the values for these variables should be read from `go env`.  This enables the configuration values to be defined in a session-independent way in the Go tool's config file.  (It's also where I personally configure my value for `GOPROXY`, so reading just the environment variable won't pick up the value I've configured.)